src/stream_compression.jl

```diff
@@ -138,7 +128,7 @@ function TranscodingStreams.process(codec::LZ4Compressor, input::Memory, output:
 
 end
 
-struct LZ4Decompressor <: TranscodingStreams.Codec
+mutable struct LZ4Decompressor <: TranscodingStreams.Codec
     dctx::Ref{Ptr{LZ4F_dctx}}
 end
```

Comment:
We should switch this back to being non-mutable. You'll be able to modify the reference still.